dataland-frontend/src/components/general/TheHowItWorks.vue

```diff
@@ -0,0 +1,281 @@
+<template>
+  <section v-if="howItWorksSection" class="howitworks" role="region" aria-labelledby="howitworks-heading">
+    <div class="howitworks__wrapper">
+      <h2 id="howitworks-heading" class="howitworks__title">
+        {{ sectionText }}
+      </h2>
+      <div ref="slider" role="list" class="howitworks__slides" @pointerdown="dragStart" @touchstart="dragStart">
+        <div v-for="(slide, index) in slides" :key="index" role="listitem" class="howitworks__slide">
+          <h3 class="howitworks__slide-title">{{ slide.title }}</h3>
+          <p class="howitworks__slide-text">{{ slide.text }}</p>
+          <p class="howitworks__slide-index">0{{ index + 1 }}</p>
+        </div>
+      </div>
+      <div class="howitworks__arrows">
+        <button
+          @click="move(-1)"
+          aria-label="Previous slide"
+          class="howitworks__arrow howitworks__arrow--left"
+        ></button>
+        <button @click="move(1)" aria-label="Next slide" class="howitworks__arrow howitworks__arrow--right"></button>
+      </div>
+    </div>
+  </section>
+</template>
+
+<script setup lang="ts">
+import { ref, computed, onUnmounted, watchEffect } from "vue";
+import type { Section } from "@/types/ContentTypes";
+
+const { sections } = defineProps<{ sections?: Section[] }>();
+const howItWorksSection = computed(() => sections?.find((s) => s.title === "How it works"));
+const sectionText = computed(() => howItWorksSection.value?.text.join(" ") ?? "");
+const slides = computed(() => sections?.find((s) => s.title === "How it works")?.cards ?? []);
+const slider = ref<HTMLElement | null>(null);
+const currentSlide = ref(0);
+
+let isDragging = false;
+let startPos = 0;
+let currentTranslate = 0;
```

Comment:
Question out of curiousity (because it is the first time I see this script syntax in Vue):

Why is currentSlide set as `ref( number)` and e.g. currentTranslate as `let currentTranslate` ?
They are both variables that are used throughout several functions in this component. What is the key difference between both, so that you have chosen to make one of those a ref?

Thanks in advance!

Reply:
The `ref` is a reactive reference that enables the Vue component to update whenever `currentSlide` changes. When you use `ref`, it ensures that any part of the template that relies on `currentSlide` will re-render when its value changes.

On the other hand, `let currentTranslate` is just a regular JavaScript variable. Changes to `currentTranslate` will not cause any part of the Vue component to re-render automatically. It is used as a temporary variable to hold the translate position while dragging, and it doesn't need to trigger any updates to the component or its template.

In summary, `currentSlide` is reactive because the component needs to react to its changes, while `currentTranslate` is non-reactive because it's managed internally within the setup function and does not affect the component's rendering directly.

Reply:
Ok got it. Thanks!